pkg/domain/infosync/placement_manager.go

```diff
@@ -107,11 +109,38 @@ func (m *mockPlacementManager) PutRuleBundles(_ context.Context, bundles []*plac
 		m.bundles = make(map[string]*placement.Bundle)
 	}
 
+	rules := 0
 	for _, bundle := range bundles {
 		if bundle.IsEmpty() {
 			delete(m.bundles, bundle.ID)
 		} else {
 			m.bundles[bundle.ID] = bundle
+			rules += len(bundle.Rules)
+		}
+	}
+
+	// Check that no bundles are overlapping
+	type keyRange struct {
+		start string
+		end   string
+	}
+	keys := make([]keyRange, 0, rules)
+	for k := range m.bundles {
```

Comment:
I think we only need to check the rules inside one bundle just now. If `GroupBundle.Override` is set to `true`, it is allowed to have multiple leader rules. 

And in one `RuleGroup` when a rule has `Override==true`, it should also be skipped.

Reply:
We still need to check the rules in one bundle even `bundle[k].Override is true`, but the current implementation skips all rules in such bundle?

Reply:
> We still need to check the rules in one bundle even `bundle[k].Override is true`

true.

Reply:
@lcwangchao @xhebox please review the new duplicate leader range check.

Reply:
I have changed the logic to only check bundles one by one, and tried to mimic the logic of PDs `prepareRulesForApply()` and `checkApplyRules()`, for any other check I created https://github.com/pingcap/tidb/issues/57693 as a follow up.